fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryRecursivePlan.java

```diff
@@ -0,0 +1,299 @@
+    }
+
+    @Nonnull
+    public Quantifier.Physical getChildQuantifier() {
+        return childQuantifier;
+    }
+
+    @SuppressWarnings("resource")
+    @Nonnull
+    @Override
+    public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final FDBRecordStoreBase<M> store,
+                                                                     @Nonnull final EvaluationContext context,
+                                                                     @Nullable final byte[] continuation,
+                                                                     @Nonnull final ExecuteProperties executeProperties) {
+
+        final var nestedExecuteProperties = executeProperties.clearSkipAndLimit();
+        return RecursiveCursor.create(
+                        rootContinuation ->
+                                rootQuantifier.getRangesOverPlan().executePlan(store, context, rootContinuation, nestedExecuteProperties),
+                        (parentResult, depth, innerContinuation) -> {
+                            // TODO: Consider binding depth as well.
+                            final EvaluationContext childContext = context.withBinding(Bindings.Internal.CORRELATION.bindingName(priorValueCorrelation.getId()), parentResult);
+                            return childQuantifier.getRangesOverPlan().executePlan(store, childContext, innerContinuation, nestedExecuteProperties);
+                        },
+                        null,
+                        continuation
+                ).skipThenLimit(executeProperties.getSkip(), executeProperties.getReturnedRowLimit())
+                .map(RecursiveCursor.RecursiveValue::getValue);
+    }
+
+    @Override
+    public int getRelationalChildCount() {
+        return 2;
+    }
+
+    @Override
+    public boolean canCorrelate() {
+        return true;
+    }
+
+    @Nonnull
+    @Override
+    public List<RecordQueryPlan> getChildren() {
+        return ImmutableList.of(rootQuantifier.getRangesOverPlan(), childQuantifier.getRangesOverPlan());
+    }
+
+    @Nonnull
+    @Override
+    public AvailableFields getAvailableFields() {
+        return AvailableFields.NO_FIELDS;
+    }
+
+    @Nonnull
+    @Override
+    public Set<CorrelationIdentifier> getCorrelatedToWithoutChildren() {
+        return resultValue.getCorrelatedTo();
+    }
+
+    @Override
+    public boolean isReverse() {
+        return Quantifiers.isReversed(Quantifiers.narrow(Quantifier.Physical.class, getQuantifiers()));
+    }
+
+    @Override
+    public boolean isStrictlySorted() {
+        return true;
+    }
+
+    @Nonnull
+    @Override
+    public Value getResultValue() {
+        return resultValue;
+    }
+
+    @Nonnull
+    @Override
+    public String toString() {
+        return ExplainPlanVisitor.toStringForDebugging(this);
+    }
+
+    @Override
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
+    public boolean equalsWithoutChildren(@Nonnull RelationalExpression otherExpression,
```

Comment:
Don't you have to worry about that prior alias here? In fact, you either have to enforce that they are the same (not as good), or override `structuralEquals` and add a mapping between the (non-equal) prion aliases before descending into the children for the equals.